base/bucket.go

```diff
@@ -119,6 +118,11 @@ type CouchbaseBucketType int
 
 // Full specification of how to connect to a bucket
 type BucketSpec struct {
+	BucketSpecOptions
```

Comment:
This refactor feels arbitrary - I don't think KvPoolSize is any more or less of an 'option' than things like BucketOpTimeout.  What's the problem that this refactor fixes?

Reply:
KvPoolSize isn't an input parameter, it is only ever read from the connection string. It is parsed from the connection string and then stored when creating a bucket to determine how to throttle kvops in the below code.

```
	numPools := 1
	if spec.KvPoolSize > 0 {
		numPools = spec.KvPoolSize
	}
	gocbv2Bucket.kvOps = make(chan struct{}, MaxConcurrentSingleOps*nodeCount*numPools)
```

A different way of addressing the immutability of the struct would be to never set this value and avoid the use of a constructor for this struct. Then, inside `GetGocbV2BucketFromCluster` you could call `getKvPoolSize(spec.Server)` which would reparse the connection string. The reason I didn't do it like this was to only parse the connection string once but this is such a cheap operation (string parsing) compared to bucket opening that it would be fine.

Do you think this would be more readable?

Reply:
I am not set on https://github.com/couchbase/sync_gateway/pull/6680/commits/2f038b936796ca9afd7557675a86ad396328f332 but it simplifies the PR.

This will behave identically to the original PR but in order for serverless values to work, you have to modify the connection string on the `BucketSpec` to have `kv_pool_size=1`. This is done in `GetBucketSpec` anyway to add `dcp_buffer_size` and `kv_buffer_size` custom values so I think this is probably a reasonable refactor.

The downsides (which aren't bad):

- the setting of an implicit `kv_pool_size` is set as `DefaultGocbKvPoolSize` inside some functions - `BucketSpec.getGoCBConnString` and `BucketSpec.GetKvPoolSize` and they have the potential to get out of sync. 
- the place that we will validate that we haven't passed `kv_pool_size` twice in a connection string will take place when opening a bucket and not loading a database. This is unlikely behavior we didn't test for anyway.
- we parse the connection string twice when creating a `Bucket`, once to get the connection string for a cluster and another for figuring out the `kv_pool_size`. This is a cheap string operation vs a kv op for a rare condition.